src/main/scala/chisel3/util/SRAM.scala

```diff
@@ -104,6 +106,24 @@ class SRAMInterface[T <: Data](
     Vec(numReadwritePorts, new MemoryReadWritePort(tpe, addrWidth, masked))
 }
 
+private[chisel3] abstract class MemoryFile(private[chisel3] val fileType: MemoryLoadFileType) {
+  val path: String
+}
```

Comment:
```suggestion
/** A memory file with which to preload an [[SRAM]]
  *
  * See concrete subclasses [[BinaryMemoryFile]] and [[HexMemoryFile]]
  */
sealed abstract class MemoryFile(private[chisel3] val fileType: MemoryLoadFileType) {
  /** The path to the memory contents file */
  val path: String
}
```

Since this type is part of the public API of SRAM, it should be public.